### PR TITLE
added a basic test case

### PR DESCRIPTION
**Pull request by kkmann**
not ready for merging!

I added our basic case, which is weirdly broken, cf. 

https://github.com/kkmann/otsd/blob/3f78546a7888e4c45d0e4c905277e4a43658a510/tests/testthat/test_examples.R#L42-L56

(just run the test code in your command line). The n() function is large right where the data distribution should place maximal weight on the conditional sample size - makes no sense at all x/

### CODE REVIEW

**Comment by kkmann** [comment]
#47 hopefully explains what is going on. I only kept the high-precision test. Beware that this test runs about 6 minutes!

**Comment by kkmann** [comment]
your tests are broken - would be best to rewind your changes (i.e. revert to #d530as7), merge that and then do your stuff in the next iteration ;) Good exercise on reverting changes anyway ;)

File: tests/testthat/test_IntegralScores.R [hunk 1]
```diff
@@ -0,0 +1,123 @@
+context("Test integral scores")
+
+test_that("Expected sample size is computed correctly",{
+    null <- PointMassPrior(.0, 1)
+    alternative <- PointMassPrior(.4, 1)
+    dist <- Normal()
+
+    # Define design from rpact
+    design_rp2 <- rpact::getDesignInverseNormal(kMax = 2, alpha = 0.05,
```
Comment (kkmann) [inline tests/testthat/test_IntegralScores.R]: indentation!

File: tests/testthat/test_IntegralScores.R [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+        w2 <- sqrt(1 - w1^2)
+        out <- (design_rp2$criticalValues[2] - w1 * z) / w2
+        return(out)
+    }
+
+    x <- GaussLegendreRule(5)$nodes
+    h <- (c1e - c1f) / 2
+    x <- h * x + (h + c1f)
+
+    design_gs <<- gq_design(round(n1),
```
Comment (kkmann) [inline tests/testthat/test_IntegralScores.R]: intendation!

File: tests/testthat/test_IntegralScores.R [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+                           c1f,
+                           c1e,
+                           rep(round(n2), 5),
+                           sapply(seq(c1f, c1e, length.out = 5), f),
+                           5L)
+
+
+    # Simulation
+    sim_alt  <<- simulate(design_gs, nsim = 10000, dist = Normal(),
```
Comment (kkmann) [inline tests/testthat/test_IntegralScores.R]: intendation

File: tests/testthat/test_IntegralScores.R [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+
+
+    # Expected Sample sizes under H1
+    expect_equal(
+        res$expectedPatientsH1 / 2, # per group!
+        evaluate(integrate(ConditionalSampleSize(dist, alternative)), design_gs),
+        tolerance = 0.5)
+
+    expect_equal(res$expectedPatientsH1 / 2,
+                 evaluate(integrate(ConditionalSampleSize(dist, alternative)),
```
Comment (kkmann) [inline tests/testthat/test_IntegralScores.R]: intendation

File: tests/testthat/test_IntegralScores.R [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+                          design_gs, specific = FALSE),
+                 tolerance = 0.5)
+
+    expect_equal(res$expectedPatientsH1 / 2,
+                 mean(sim_alt[, "n1"]) + mean(sim_alt[,"n2"]),
```
Comment (kkmann) [inline tests/testthat/test_IntegralScores.R]: same...

File: tests/testthat/test_IntegralScores.R [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+
+
+    # Expected Sample sizes under H0
+    expect_equal(
+        res$expectedPatientsH0 / 2, # per group!
+        evaluate(integrate(ConditionalSampleSize(dist, null)), design_gs),
+        tolerance = 0.5)
+
+    expect_equal(res$expectedPatientsH0 / 2,
+                 evaluate(integrate(ConditionalSampleSize(dist, null)),
```
Comment (kkmann) [inline tests/testthat/test_IntegralScores.R]: more intendation, check code guidelines!

File: tests/testthat/test_IntegralScores.R [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+                          design_gs, specific = FALSE),
+                 tolerance = 0.5)
+
+    expect_equal(res$expectedPatientsH0 / 2,
+                 mean(sim_null[, "n1"]) + mean(sim_null[,"n2"]),
```
Comment (kkmann) [inline tests/testthat/test_IntegralScores.R]: intendation

File: tests/testthat/test_IntegralScores.R [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+        tolerance = .01)
+
+    expect_equal(
+        mean(sim_alt[, "reject"]),
+        .8,
+        tolerance = .01)
+}) # end 'power is computed correctly'
+
+
+test_that("Type one error is computed correctly", {
```
Comment (kkmann) [inline tests/testthat/test_IntegralScores.R]: this is just checked for a single desinǵn, be more precise plz

File: tests/testthat/test_IntegralScores.R [hunk 9]
```diff
@@ -0,0 +1,123 @@
...
+                 tolerance = 0.5)
+
+    expect_equal(res$expectedPatientsH0 / 2,
+                 mean(sim_null[, "n1"]) + mean(sim_null[,"n2"]),
+                 tolerance = 0.5)
+
+}) # end 'expected sample size is computed correctly'
+
+
+test_that("Power is computed correctly", {
```
Comment (kkmann) [inline tests/testthat/test_IntegralScores.R]: this is just for one specific design, be more specific

File: tests/testthat/test_IntegralScores.R [hunk 10]
```diff
@@ -0,0 +1,123 @@
+context("Test integral scores")
+
+test_that("Expected sample size is computed correctly",{
```
Comment (kkmann) [inline tests/testthat/test_IntegralScores.R]: very difficult to follow - could you maybe comment a bit on what you ACTUALLY want to do?

File: tests/testthat/test_examples.R [hunk 11]
```diff
@@ -0,0 +1,52 @@
+context("test point alternative point ESS design (case 01)")
+
+test_that("plausible n2() function", {
+
+    order <- 9L
```
Comment (kkmann) [inline tests/testthat/test_examples.R]: if you want, you could scale the test down to 7L - maybe it converges faster then

**Comment by kkmann** [comment]
@MatheMax, you do know that you can run individual test files in the new rstudio preview, right? https://www.rstudio.com/products/rstudio/download/preview/

**Comment by MatheMax** [comment]
> 
> 
> @MatheMax, you do know that you can run individual test files in the new rstudio preview, right? https://www.rstudio.com/products/rstudio/download/preview/

Now I do ;)

File: tests/testthat/test_GQDesign.R [hunk 12]
```diff
@@ -121,4 +121,69 @@ test_that("Optimal design with point prior is computable", {
...
+    c1f <- qnorm(char$futilityProbabilities) +
+        sqrt(res$numberOfPatientsGroup1[1]) * .3 / sqrt(2)
+    c1e <- design_rp$criticalValues[1]
+
+    x <- GaussLegendreRule(5)$nodes
+    h <- (c1e - c1f) / 2
+    x <- h * x + (h + c1f)
+
+    design_gs <- gq_design(ceiling(n1),
+                           c1f,
```
Comment (kkmann) [inline tests/testthat/test_GQDesign.R]: ...

File: tests/testthat/test_GQDesign.R [hunk 13]
```diff
@@ -121,4 +121,69 @@ test_that("Optimal design with point prior is computable", {
...
 
 
+}) # end 'Optimal design with point prior is computable'
+
+
+test_that("Optimal design is superior to standard GS design", {
+
+    # Create design from rpact
+    design_rp <- rpact::getDesignInverseNormal(kMax = 2, alpha = 0.025, beta = 0.2,
+                                            futilityBounds = 0,
```
Comment (kkmann) [inline tests/testthat/test_GQDesign.R]: ...

File: tests/testthat/test_GQDesign.R [hunk 14]
```diff
@@ -121,4 +121,69 @@ test_that("Optimal design with point prior is computable", {
...
+                                            typeOfDesign = "P")
+
+    res       <- rpact::getSampleSizeMeans(design_rp, normalApproximation = TRUE,
+                                           alternative = .3)
```
Comment (kkmann) [inline tests/testthat/test_GQDesign.R]: ...

**Comment by MatheMax** [comment]
@kkmann Do you understand why the first check did not work? Because I removed the "known-values"-file?

**Comment by kkmann** [comment]
nope, needed to restart travis (busy today ;)). Fine by me, merging.

**Pull request merged**